### ISSUE TITLE
Support for ndx-events.Events and pynwb.behavior.BehavioralEpochs

### ISSUE THREAD

**Issue by pauladkisson** [flatironinstitute/neurosift#168]
See https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7d1990d7-a714-406e-9cb2-1b8f4015869e/download/&dandisetId=000971&dandisetVersion=draft for events and https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7d1990d7-a714-406e-9cb2-1b8f4015869e/download/&dandisetId=000971&dandisetVersion=draft for BehavioralEpochs

**Comment by pauladkisson** [comment]
Has there been any progress on this? Are there any blockers I can help with?

Also a note: `ndx_events.Events` is specifically from version 0.2.0

**Comment by magland** [comment]
Hi @pauladkisson I've been working on some other things so probably won't get to this for a little while. Do you know typescript/react? I can point you to the relevant code if you want to take a look.

**Comment by pauladkisson** [comment]
> Do you know typescript/react? I can point you to the relevant code if you want to take a look.

Unfortunately I do not.

But thank you for the update.

**Comment by magland** [comment]
Could you describe what the views should look like?

For processing/behavior/behavioral_epochs/reward_port_intervals
We have the IntervalSeries view:
https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7d1990d7-a714-406e-9cb2-1b8f4015869e/download/&dandisetId=000971&dandisetVersion=draft&tab=neurodata-item:/processing/behavior/behavioral_epochs/reward_port_intervals|IntervalSeries

**Comment by pauladkisson** [comment]
I would suggest an analog to [matplotlib.pyplot.eventplot](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.eventplot.html) for the events and filled region for the IntervalSeries.

For example, see my [example notebook](https://github.com/dandi/example-notebooks/blob/cc03f6cda9672b0f9575f2f711fd4e9a915ad93b/000971/lernerlab/seiler_2024/optogenetics_example_notebook.ipynb)

**Comment by pauladkisson** [comment]
Bumping this bc my current conversion also needs ndx-events.Events visualization: https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7f969b84-db9b-48ab-8935-b1ca21f5b607/download/&dandisetId=001259&dandisetVersion=draft

**Comment by magland** [comment]
@pauladkisson I think that Dandiset is embargoed. Would you be able to give me access?

**Comment by pauladkisson** [comment]
Yes, you are right it is embargoed.  You should have access now.

**Comment by magland** [comment]
@pauladkisson I have added a simple view plugin for the ndx-events/Events type.

https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7f969b84-db9b-48ab-8935-b1ca21f5b607/download/&dandisetId=001259&dandisetVersion=draft&tab=neurodata-items:neurodata-item:/processing/behavior/targetOUT|Events@neurodata-item:/processing/behavior/toneIN|Events@neurodata-item:/processing/behavior/toneOUT|Events@neurodata-item:/processing/behavior/valve|Events

Let me know what the next iteration should be.

**Comment by pauladkisson** [comment]
Events look good to me!

**Comment by pauladkisson** [comment]
Not strictly related to this issue, but what is going on with the encoder time series plot: https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/7f969b84-db9b-48ab-8935-b1ca21f5b607/download/&dandisetId=001259&dandisetVersion=draft&tab=neurodata-items:neurodata-item:/processing/behavior/targetOUT%7CEvents@neurodata-item:/processing/behavior/toneIN%7CEvents@neurodata-item:/processing/behavior/toneOUT%7CEvents@neurodata-item:/processing/behavior/valve%7CEvents

I'm not seeing any time series...

**Comment by magland** [comment]
I think it's probably due to some NaN's that NS is choking on. Perhaps open a separate issue for that?

**Comment by pauladkisson** [comment]
See new issue here: #219

**Comment by pauladkisson** [comment]
So ndx-events.Events works now, but pynwb.behavior.BehavioralEpochs is still broken.  Would you like me to close this issue and open up a new one just for pynwb.behavior.BehavioralEpochs?

**Comment by magland** [comment]
> So ndx-events.Events works now, but pynwb.behavior.BehavioralEpochs is still broken. Would you like me to close this issue and open up a new one just for pynwb.behavior.BehavioralEpochs?

That would be helpful thanks.

**Comment by pauladkisson** [comment]
See new issue: #220
